samples/Ark.ReferenceProject/Core/Ark.Reference.Core.WebInterface/Startup.cs

```diff
@@ -56,6 +57,27 @@ public override void ConfigureServices(IServiceCollection services)
         {
             base.ConfigureServices(services);
 
+            // Configure System.Text.Json source generation with Ark defaults
+            // Using chained resolver to combine source generation with reflection fallback
+            // This avoids naming collisions for auto-generated nested types like IEnumerable<T>
+            // See: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation
+            var jsonOptions = new System.Text.Json.JsonSerializerOptions();
+            System.Text.Json.Extensions.ConfigureArkDefaults(jsonOptions);
+            
+            var jsonContext = new CoreApiJsonSerializerContext(jsonOptions);
+
+            services.ConfigureHttpJsonOptions(options =>
+            {
+                options.SerializerOptions.TypeInfoResolverChain.Insert(0, jsonContext);
```

Comment:
remove the Reflection fallback. Assume that only source generated are supported.

options.SerializeOptions.TypeInfoResolver = jsonContext;